snorkel/labeling/model/wrapper.py

```diff
@@ -0,0 +1,191 @@
+    lr_scheduler
+        Which lr_scheduler to use (one of ["constant", "linear", "exponential", "step"])
+    lr_scheduler_config
+        Settings for the LRScheduler
+    prec_init
+        LF precision initializations / priors
+    seed
+        A random seed to initialize the random number generator with
+    log_freq
+        Report loss every this many epochs (steps)
+    mu_eps
+        Restrict the learned conditional probabilities to [mu_eps, 1-mu_eps]
+    """
+
+    def __init__(
+        self,
+        L_train: np.ndarray,
+        cardinality: int = 2,
+        verbose: bool = True,
+        device: str = "cpu",
+        metric: str = "accuracy",
+        tie_break_policy: str = "abstain",
+        n_epochs: int = 100,
+        lr: float = 0.01,
+        l2: float = 0.0,
+        optimizer: str = "sgd",
+        optimizer_config: Optional[OptimizerConfig] = None,
+        lr_scheduler: str = "constant",
+        lr_scheduler_config: Optional[LRSchedulerConfig] = None,
+        prec_init: float = 0.7,
+        seed: int = np.random.randint(1e6),
+        log_freq: int = 10,
+        mu_eps: Optional[float] = None,
+        class_balance: Optional[List[float]] = None,
+        **kwargs: Any,
+    ) -> None:
+
+        self.L_train = L_train
+        self.cardinality = cardinality
+        self.verbose = verbose
+        self.device = device
+        self.metric = metric
+        self.tie_break_policy = tie_break_policy
+        self.n_epochs = n_epochs
+        self.lr = lr
+        self.l2 = l2
+        self.optimizer = optimizer
+        self.optimizer_config = (
+            OptimizerConfig()  # type: ignore
+            if optimizer_config is None
+            else optimizer_config
+        )
+        self.lr_scheduler = lr_scheduler
+        self.lr_scheduler_config = (
+            LRSchedulerConfig()  # type: ignore
+            if lr_scheduler_config is None
+            else lr_scheduler_config
+        )
+        self.prec_init = prec_init
+        self.seed = seed
+        self.log_freq = log_freq
+        self.mu_eps = mu_eps
+        self.class_balance = class_balance
+
+        self.label_model = LabelModel(
+            cardinality=self.cardinality, verbose=self.verbose, device=self.device
+        )
+
+    def fit(
+        self, L_dev: np.ndarray, Y_dev: Optional[np.ndarray] = None
+    ) -> "SKlearnLabelModel":
+        """
+        Train label model.
+
+        **Note that one of the inputs is L_dev but it is not used to train the LabelModel. This is to make the wrapper compatible with scikit-learn hyperparameter search functionality.
+
+        Parameters
+        ----------
+        L_dev
+             An [n,m] matrix with values in {-1,0,1,...,k-1}
+        Y_dev
+             Gold labels associated with data points in L_dev
+
+        Returns
+        -------
+        SKlearnLabelModel
+        """
+        self.label_model.fit(
+            L_train=self.L_train,
+            Y_dev=Y_dev,
```

Comment:
do we need to provide Y_dev here to use CV? i think no, just for print outs

Reply:
We need Y_dev because that’s what gridsearch expects, without it, it thinks it’s a unsupervised learning alg